fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/models/BinaryImpl.java

```diff
@@ -135,7 +137,11 @@ public String getFilename() {
     @Override
     public FedoraResource getDescription() {
         try {
-            final var descId = getId() + "/" + FCR_METADATA;
+            final FedoraID descId = getFedoraId().addToResourceId(FCR_METADATA);
+            if (this.isMemento()) {
+                return resourceFactory.getResource(tx, descId.addToFullId(FCR_VERSIONS + "/" +
```

Comment:
The `addTo` methods could also be changed to have signatures like `FedoarId addTo...(String part, String... parts)`